### DISCUSSION TITLE
Wrong type returned when using Alias

### DISCUSSION THREAD

**Discussion by Khalil-Bouzekri** [LemonAppDev/konsist#1131] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.15.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

With the following classes:

```
package com.test.konsist.therealone

class A(val z: Boolean) 
```

```
package com.test.konsist.iamthealias

class A(val x: Int) 
```

```
package com.test.konsist

import com.test.konsist.iamthealias.A as OtherA
import com.test.konsist.therealone.A

class B {
  fun toto(): A = A(OtherA(0).x == 0)
} 
```

And the following Konsist test:

```
package com.test.konsist

import com.lemonappdev.konsist.api.Konsist
import com.lemonappdev.konsist.api.ext.list.withPackage
import com.lemonappdev.konsist.api.ext.list.withReturnType
import org.junit.jupiter.api.Test

class KonsistTest {

  @Test
  fun test() {
    Konsist.scopeFromTest()
      .functions()
      .withReturnType()
      .withPackage("com.test.konsist")
      .forEach { println(it.returnType!!.declaration.packagee!!.fullyQualifiedName) }
  }
}
```

The class under the alias is returned instead of the correct one.

Note that if the 2 imports are swapped like this it works:

```
import com.test.konsist.therealone.A
import com.test.konsist.iamthealias.A as OtherA
```

But it is a problem as we use a global formatter and cannot change the order of imports.

### Actual behaviour

`com.test.konsist.iamthealias`

### Expected behaviour

`com.test.konsist.therealone`

**Comment by nataliapeterwas** [comment, marked as answer]
Thank you for reporting the problem.
We will fix this issue in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1354)

**Comment by nataliapeterwas** [comment]
Konsist `0.17.0` has been released! This release contains a fix for this issue :) See full release log: [Release log](https://github.com/LemonAppDev/konsist/releases/tag/v0.17.0)
